lib/_stream_readable.js

```diff
@@ -447,7 +447,7 @@ function maybeReadMore_(stream, state) {
 // for virtual (non-string, non-buffer) streams, "length" is somewhat
 // arbitrary, and perhaps not very meaningful.
 Readable.prototype._read = function(n) {
-  this.emit('error', new Error('not implemented'));
+  process.nextTick(() => this.emit('error', new Error('not implemented')));
 };
```

Comment:
Same here... please create the Error object first, *then* dispatch to process.nextTick()...